torch_geometric/transforms/feature_prop.py

```diff
@@ -0,0 +1,83 @@
+import torch
+from torch_sparse import SparseTensor, matmul
+
+from torch_geometric.data import Data
+from torch_geometric.data.datapipes import functional_transform
+from torch_geometric.nn.conv.gcn_conv import gcn_norm
+from torch_geometric.transforms import BaseTransform
+from torch_geometric.typing import OptTensor
+
+
+@functional_transform('feature_propagation')
+class FeaturePropagation(BaseTransform):
+    r"""The feature propagation operator from the
+    `"On the Unreasonable Effectiveness of Feature
+    propagation in Learning on Graphs with Missing Node Features"
+    <https://arxiv.org/abs/2111.12128>`_ paper
+
+    .. math::
+        \mathbf{X}^{\prime} = \mathbf{D}^{-1/2} \mathbf{A}
+        \mathbf{D}^{-1/2} \mathbf{X},
+
+        \mathbf{X}_{k} = \mathbf{X}^{\prime}_{k}
+
+
+    where missing node features are inferred by known features via propagation.
+
+    .. note::
+
+        For an example of using :class:`~torch_geometric.transforms.
+        FeaturePropagation`, see `examples/feature_propagation.py
+        <https://github.com/pyg-team/pytorch_geometric/blob/master/examples/
+        feature_propagation.py>`_.
```

Comment:
Not a strong preference. But i don't think we need an example for this, a code snippet with the documentation will do, like done [here](https://pytorch-geometric.readthedocs.io/en/latest/modules/transforms.html#torch_geometric.transforms.RandomLinkSplit).

Reply:
Indeed. Thanks for your suggestion!